{
  "comments": [
    "When asking JUnit to execute all Test in a given dir, \nTestCases extending org.custommonkey.xmlunit.XMLTestCase  cause the Eclipse Junit Runner to incorrectly try to test org.custommonkey.xmlunit.XMLTestCase\n\nHere is the stack trace.\n\njunit.framework.AssertionFailedError: No tests found in org.custommonkey.xmlunit.XMLTestCase\n\tat junit.framework.Assert.fail(Assert.java:47)\n\tat junit.framework.TestSuite$1.runTest(TestSuite.java:263)\n\tat junit.framework.TestCase.runBare(TestCase.java:127)\n\tat junit.framework.TestResult$1.protect(TestResult.java:106)\n\tat junit.framework.TestResult.runProtected(TestResult.java:124)\n\tat junit.framework.TestResult.run(TestResult.java:109)\n\tat junit.framework.TestCase.run(TestCase.java:118)\n\tat junit.framework.TestSuite.runTest(TestSuite.java:208)\n\tat junit.framework.TestSuite.run(TestSuite.java:203)\n\tat org.eclipse.jdt.internal.junit.runner.junit3.JUnit3TestReference.run(JUnit3TestReference.java:128)\n\tat org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:460)\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:673)\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:386)\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:196)\n\nI will attach a very small eclipse project that demonstrates this.",
    "Created an attachment (id\u003d53778)\nTest Project to show bug.\n\nDownload XMLunit from http://sourceforge.net/project/showfiles.php?group_id\u003d23187\u0026package_id\u003d15921\u0026release_id\u003d155097\nand place the jar in the lib dir",
    "This got fixed in 3.3. As a workaround, to avoid the warning, create a TestSuite that references your tests. Running a project or package with JUnit has several problem (for example repeated tests) that we can\u0027t really solve.",
    "Changing OS from Mac OS to Mac OS X as per bug 185991"
  ],
  "commentCreationDates": [
    "2006-11-13T19:37:49+01:00",
    "2006-11-13T21:21:22+01:00",
    "2006-11-15T14:47:40+01:00",
    "2007-07-29T15:21:07+02:00"
  ],
  "traces": [
    {
      "exceptionType": "junit.framework.AssertionFailedError",
      "message": "No tests found in org.custommonkey.xmlunit.XMLTestCase",
      "elements": [
        {
          "method": "junit.framework.Assert.fail",
          "source": "Assert.java:47"
        },
        {
          "method": "junit.framework.TestSuite$1.runTest",
          "source": "TestSuite.java:263"
        },
        {
          "method": "junit.framework.TestCase.runBare",
          "source": "TestCase.java:127"
        },
        {
          "method": "junit.framework.TestResult$1.protect",
          "source": "TestResult.java:106"
        },
        {
          "method": "junit.framework.TestResult.runProtected",
          "source": "TestResult.java:124"
        },
        {
          "method": "junit.framework.TestResult.run",
          "source": "TestResult.java:109"
        },
        {
          "method": "junit.framework.TestCase.run",
          "source": "TestCase.java:118"
        },
        {
          "method": "junit.framework.TestSuite.runTest",
          "source": "TestSuite.java:208"
        },
        {
          "method": "junit.framework.TestSuite.run",
          "source": "TestSuite.java:203"
        },
        {
          "method": "org.eclipse.jdt.internal.junit.runner.junit3.JUnit3TestReference.run",
          "source": "JUnit3TestReference.java:128"
        },
        {
          "method": "org.eclipse.jdt.internal.junit.runner.TestExecution.run",
          "source": "TestExecution.java:38"
        },
        {
          "method": "org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests",
          "source": "RemoteTestRunner.java:460"
        },
        {
          "method": "org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests",
          "source": "RemoteTestRunner.java:673"
        },
        {
          "method": "org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run",
          "source": "RemoteTestRunner.java:386"
        },
        {
          "method": "org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main",
          "source": "RemoteTestRunner.java:196"
        }
      ],
      "number": 0,
      "commentIndex": 0,
      "bugId": "164377",
      "date": "2006-11-13T19:37:49+01:00",
      "product": "JDT",
      "component": "UI",
      "severity": "normal"
    }
  ],
  "groupId": "164377",
  "bugId": "164377",
  "date": "2006-11-13T19:37:49+01:00",
  "product": "JDT",
  "component": "UI",
  "severity": "normal"
}